lib/App.js

```diff
@@ -358,3 +358,21 @@ App.prototype._handleMessage = function(action, message) {
     });
   }
 };
+
+function parseState(innerHtml) {
```

Comment:
Let's export this as a "static function" so we can write a few small tests for it. Also, how about a name like `_parseInitialData` to be clearer about its function?

Something like:
```
App._parseInitialData = function _parseInitialData(jsonString) {
```